conda_lock/src_parser/aggregation.py

```diff
@@ -51,3 +54,34 @@ def aggregate_lock_specs(
             lock_spec.allow_pypi_requests for lock_spec in lock_specs
         ),
     )
+
+
+def unify_package_sources(collections: List[List[Channel]]) -> List[Channel]:
+    """Unify the package sources from multiple lock specs.
+
+    To be able to merge the lock specs, the package sources must be compatible between
+    them. This means that between any two lock specs, the package sources must be
+    identical or one must be an extension of the other.
+
+    This allows us to use a superset of all of the package source lists in the
+    aggregated lock spec.
+
+    The following is allowed:
+
+    > unify_package_sources([[channel_two, channel_one], [channel_one]])
+    [channel_two, channel_one]
+
+    Whilst the following will fail:
+
+    > unify_package_sources([[channel_two, channel_one], [channel_three, channel_one]])
+
+    In the failing example, it is not possible to predictably decide which channel
+    to search first, `channel_two` or `channel_three`, so we error in this case.
+    """
+    if not collections:
+        return []
+    result = max(collections, key=len)
+    for collection in collections:
+        if collection and collection != result[-len(collection) :]:
+            raise ValueError(f"{collection} is not an ordered subset of {result}")
```

Comment:
I like code that I don't have to think about. While not as slick as your solution, I think it's a bit more explicit.

Also, I think the `ValueError` message was wrong (missing "at the end of").

```suggestion
        if collection == []:
            truncated_result = []
        else:
            truncated_result = result[-len(collection) :]
        if collection != truncated_result:
            raise ValueError(
                f"{collection} is not an ordered subset at the end of {result}"
            )
```